beacon-chain/p2p/discovery.go

```diff
@@ -47,6 +52,88 @@ type quicProtocol uint16
 // quicProtocol is the "quic" key, which holds the QUIC port of the node.
 func (quicProtocol) ENRKey() string { return "quic" }
 
+type listenerWrapper struct {
+	*sync.RWMutex
+	listener        *discover.UDPv5
+	listenerCreator func() (*discover.UDPv5, error)
+}
+
+func NewListener(listenerCreator func() (*discover.UDPv5, error)) (*listenerWrapper, error) {
+	rawListener, err := listenerCreator()
+	if err != nil {
+		return nil, err
+	}
+	return &listenerWrapper{
+		RWMutex:         new(sync.RWMutex),
```

Comment:
No need for `new(sync.RWMutex)`. [Zero-value Mutexed are Valid](https://github.com/uber-go/guide/blob/master/style.md#zero-value-mutexes-are-valid)